app/internal/features.py

```diff
@@ -0,0 +1,167 @@
+from fastapi import Depends
+
+from app.database.models import UserFeature, Feature
+from app.dependencies import get_db, SessionLocal
+from app.features.index import features, icons
+from app.internal.utils import create_model, get_current_user
+
+
+def create_features_at_startup(session: SessionLocal) -> bool:
+    for feat in features:
+        if not is_feature_exists_in_db(feature=feat, session=session):
+            icon = icons.get(feat['name'])
+            create_feature(**feat, icon=icon, db=session)
+
+    return True
+
+
+def is_association_exists_in_db(form: dict, session: SessionLocal) -> bool:
+    db_association = session.query(UserFeature).filter_by(
+        feature_id=form['feature_id'],
+        user_id=form['user_id']
+    ).first()
+
+    return db_association is not None
+
+
+def delete_feature(
+    feature: Feature, session: SessionLocal = Depends(get_db)
+) -> None:
+    session.query(UserFeature).filter_by(feature_id=feature.id).delete()
+    session.query(Feature).filter_by(id=feature.id).delete()
+    session.commit()
+
+
+def is_feature_exists_in_db(feature: dict, session: SessionLocal) -> bool:
+    db_feature = session.query(Feature).filter(
+        (Feature.name == feature['name']) |
+        (Feature.route == feature['route'])).first()
+
+    if db_feature is None:
+        return False
+
+    # Update if found
+    update_feature(
+        feature=db_feature, new_feature_obj=feature, session=session)
+    return True
+
+
+def update_feature(feature: Feature, new_feature_obj: dict,
```

Comment:
could you elaborate about new_feature_obj type - Dict of what?

Reply:
A `dict` object.
It represents a feature in `dict` style.

Reply:
Please if you have questions about the features panel ask them in the features panel PR, this is a icons PR.